google/cloud/pubsub/internal/pull_ack_handler_factory.cc

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/pull_ack_handler_factory.h"
+#include "google/cloud/pubsub/internal/tracing_pull_ack_handler.h"
+#include "google/cloud/internal/opentelemetry.h"
+#include "google/cloud/options.h"
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+pubsub::PullAckHandler CreatePullAckHandler(
+    CompletionQueue cq, std::weak_ptr<SubscriberStub> stub,
+    pubsub::Subscription const& subscription, std::string const& ack_id,
```

Comment:
`std::move()` for a `T const&` does not really help.  This function should probably be:

```suggestion
    pubsub::Subscription subscription, std::string  ack_id,
```